pkg/common/common_test.go

```diff
@@ -250,6 +250,274 @@ func TestContains(t *testing.T) {
+			slice1:   []string{"test-gw1", "test-gw2"},
+			slice2:   []string{"test-gw3", "test-gw4"},
+			expected: false,
+		},
+		{
+			name:     "when slice1 is empty then return false",
+			slice1:   []string{},
+			slice2:   []string{"test-gw3", "test-gw4"},
+			expected: false,
+		},
+		{
+			name:     "when slice2 is empty then return false",
+			slice1:   []string{"test-gw1", "test-gw2"},
+			slice2:   []string{},
+			expected: false,
+		},
+		{
+			name:     "when both slices are empty then return false",
+			slice1:   []string{},
+			slice2:   []string{},
+			expected: false,
+		},
+		{
+			name:     "when slice1 is nil then return false",
+			slice2:   []string{"test-gw3", "test-gw4"},
+			expected: false,
+		},
+		{
+			name:     "when slice2 is nil then return false",
+			slice1:   []string{"test-gw1", "test-gw2"},
+			expected: false,
+		},
+		{
+			name:     "when both slices are nil then return false",
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if Intersect(tc.slice1, tc.slice2) != tc.expected {
+				t.Errorf("when slice1=%v and slice2=%v, expected=%v, but got=%v", tc.slice1, tc.slice2, tc.expected, !tc.expected)
+			}
+		})
+	}
+}
+
+func TestIntersectWithInts(t *testing.T) {
+	testCases := []struct {
+		name     string
+		slice1   []int
+		slice2   []int
+		expected bool
+	}{
+		{
+			name:     "when slice1 and slice2 have one common item then return true",
+			slice1:   []int{1, 2},
+			slice2:   []int{1, 3, 4},
+			expected: true,
+		},
+		{
+			name:     "when slice1 and slice2 have no common item then return false",
+			slice1:   []int{1, 2},
+			slice2:   []int{3, 4},
+			expected: false,
+		},
+		{
+			name:     "when slice1 is empty then return false",
+			slice1:   []int{},
+			slice2:   []int{3, 4},
+			expected: false,
+		},
+		{
+			name:     "when slice2 is empty then return false",
+			slice1:   []int{1, 2},
+			slice2:   []int{},
+			expected: false,
+		},
+		{
+			name:     "when both slices are empty then return false",
+			slice1:   []int{},
+			slice2:   []int{},
+			expected: false,
+		},
+		{
+			name:     "when slice1 is nil then return false",
+			slice2:   []int{3, 4},
+			expected: false,
+		},
+		{
+			name:     "when slice2 is nil then return false",
+			slice1:   []int{1, 2},
+			expected: false,
+		},
+		{
+			name:     "when both slices are nil then return false",
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if Intersect(tc.slice1, tc.slice2) != tc.expected {
+				t.Errorf("when slice1=%v and slice2=%v, expected=%v, but got=%v", tc.slice1, tc.slice2, tc.expected, !tc.expected)
+			}
+		})
+	}
+}
+
+func TestIntersection(t *testing.T) {
+	testCases := []struct {
+		name     string
+		slice1   []string
+		slice2   []string
+		expected []string
+	}{
+		{
+			name:     "when slice1 and slice2 have one common item then return true",
```

Comment:
nitpick: the return value is not a `bool`

Reply:
Nice catch. It was copied and pasted from the tests written for another function, `Intersect`. I'll fix it.

Reply:
Done